Makefile

```diff
@@ -906,58 +906,57 @@ finish: check-symbols
 endif
 endif
 
-DEFINED_SYMBOLS = $(SYSROOT_SHARE)/defined-symbols.txt
-UNDEFINED_SYMBOLS = $(SYSROOT_SHARE)/undefined-symbols.txt
+install: finish
+	mkdir -p "$(INSTALL_DIR)"
+	cp -r "$(SYSROOT)/lib" "$(SYSROOT)/share" "$(SYSROOT)/include" "$(INSTALL_DIR)"
 
-ifeq ($(WASI_SNAPSHOT),p2)
-EXPECTED_TARGET_DIR = expected/wasm32-wasip2
-else
-ifeq ($(THREAD_MODEL),posix)
-EXPECTED_TARGET_DIR = expected/wasm32-wasip1-threads
-else
-EXPECTED_TARGET_DIR = expected/wasm32-wasip1
-endif
-endif
 
+##### CHECK ####################################################################
+# The `check-symbols` target builds up a set of text files in `<sysroot>/share`
+# which are compared with known-good output in the `expected` directory.
+################################################################################
 
-check-symbols: startup_files libc
-	#
-	# Collect metadata on the sysroot and perform sanity checks.
-	#
+$(SYSROOT_SHARE)/defined-symbols.txt: startup_files libc
 	mkdir -p "$(SYSROOT_SHARE)"
-
-	#
-	# Collect symbol information.
-	#
+	"$(NM)" --defined-only \
+	    $(SYSROOT_LIB)/libc.a $(SYSROOT_LIB)/libwasi-emulated-*.a $(SYSROOT_LIB)/*.o \
```

Comment:
@sunfishcode: one thing that doesn't make a ton of sense is why we collect the defined symbols for `libc.a` and `libwasi-emulated-*.a` but then when we collect the undefined symbols we switch up which files we're looking at: `libc.a` and `libc-*.a`. I would have expected these to be the same files in both cases but that results in slightly different `undefined-symbols.txt` (e.g., `__floatunsitf`).

Reply:
We don't want to just check `*.a` and `*.o` in both places?

Reply:
That's a good point. I can't think of a reason offhand why they're different.